### ISSUE TITLE
Update KotlinForForge for 1.19.2 Support

### ISSUE THREAD

**Issue by Jayryn** [thedarkcolour/KotlinForForge#46]
Would you please update KotlinForForge to support latest versions of Forge 1.19.2?

**Forge Version:** 43.0.2

**Comment by thedarkcolour** [comment]
Should already work for that version. Is there a problem?

**Comment by Jayryn** [comment]
> Should already work for that version. Is there a problem?

I've had this crash when using the Inventory Profiles Next Mod (seems to be KotlinForForge Related for my eyes)

```
-- Head --
Thread: Render thread
Stacktrace:
	at thedarkcolour.kotlinforforge.KotlinModContainer.<init>(KotlinModContainer.kt:40) ~[kotlinforforge-3.6.0-obf.jar%23148!/:3.6.0] {}
-- MOD inventoryprofilesnext --
Details:
	Caused by 0: java.lang.reflect.InvocationTargetException
		at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?] {}
		at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77) ~[?:?] {}
		at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?] {}
		at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) ~[?:?] {}
		at java.lang.reflect.Constructor.newInstance(Constructor.java:480) ~[?:?] {}
		at thedarkcolour.kotlinforforge.KotlinLanguageProvider$KotlinModTarget.loadMod(KotlinLanguageProvider.kt:46) ~[kotlinforforge-3.6.0-obf.jar%23148!/:3.6.0] {}
		at net.minecraftforge.fml.ModLoader.buildModContainerFromTOML(ModLoader.java:276) ~[fmlcore-1.19.2-43.0.2.jar%23152!/:?] {}
		at net.minecraftforge.fml.ModLoader.lambda$buildMods$29(ModLoader.java:255) ~[fmlcore-1.19.2-43.0.2.jar%23152!/:?] {}
		at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?] {}
		at java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1850) ~[?:?] {}
		at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?] {}
		at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?] {}
		at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?] {}
		at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?] {}
		at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[?:?] {}
		at net.minecraftforge.fml.ModLoader.buildMods(ModLoader.java:257) ~[fmlcore-1.19.2-43.0.2.jar%23152!/:?] {}
		at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?] {}
		at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?] {}
		at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625) ~[?:?] {}
		at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?] {}
		at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?] {}
		at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575) ~[?:?] {}
		at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260) ~[?:?] {}
		at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616) ~[?:?] {}
		at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622) ~[?:?] {}
		at java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627) ~[?:?] {}
		at net.minecraftforge.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:175) ~[fmlcore-1.19.2-43.0.2.jar%23152!/:?] {}
		at net.minecraftforge.client.loading.ClientModLoader.lambda$begin$1(ClientModLoader.java:91) ~[forge-1.19.2-43.0.2-universal.jar%23156!/:?] {re:classloading,pl:runtimedistcleaner:A}
		at net.minecraftforge.client.loading.ClientModLoader.lambda$createRunnableWithCatch$4(ClientModLoader.java:111) ~[forge-1.19.2-43.0.2-universal.jar%23156!/:?] {re:classloading,pl:runtimedistcleaner:A}
		at net.minecraftforge.client.loading.ClientModLoader.begin(ClientModLoader.java:91) ~[forge-1.19.2-43.0.2-universal.jar%23156!/:?] {re:classloading,pl:runtimedistcleaner:A}
		at net.minecraft.client.Minecraft.<init>(Minecraft.java:468) ~[client-1.19.2-20220805.130853-srg.jar%23151!/:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:rubidium.mixins.json:core.MixinMinecraftClient,pl:mixin:APP:mixins.ipnext.json:MixinMinecraftClient,pl:mixin:A,pl:runtimedistcleaner:A}
		at net.minecraft.client.main.Main.m_239872_(Main.java:176) ~[client-1.19.2-20220805.130853-srg.jar%23151!/:?] {re:classloading,pl:runtimedistcleaner:A}
		at net.minecraft.client.main.Main.main(Main.java:51) ~[client-1.19.2-20220805.130853-srg.jar%23151!/:?] {re:classloading,pl:runtimedistcleaner:A}
		at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
		at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
		at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
		at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?] {}
		at net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.lambda$launchService$0(CommonClientLaunchHandler.java:27) ~[fmlloader-1.19.2-43.0.2.jar%2395!/:?] {}
		at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) [modlauncher-10.0.8.jar%2382!/:?] {}
		at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-10.0.8.jar%2382!/:?] {}
		at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-10.0.8.jar%2382!/:?] {}
		at cpw.mods.modlauncher.Launcher.run(Launcher.java:106) [modlauncher-10.0.8.jar%2382!/:?] {}
		at cpw.mods.modlauncher.Launcher.main(Launcher.java:77) [modlauncher-10.0.8.jar%2382!/:?] {}
		at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-10.0.8.jar%2382!/:?] {}
		at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-10.0.8.jar%2382!/:?] {}
		at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) [bootstraplauncher-1.1.2.jar:?] {}

	Mod File: /D:/.Games/Minecraft/.minecraft_forge_1.19/mods/InventoryProfilesNext-forge-1.19-1.6.3.jar
	Failure message: Inventory Profiles Next has class loading errors
		java.lang.reflect.InvocationTargetException: null
	Mod Version: 1.6.3
	Mod Issue URL: https://github.com/blackd/Inventory-Profiles/issues
	Exception message: java.lang.IncompatibleClassChangeError: Method 'net.minecraftforge.eventbus.api.BusBuilder net.minecraftforge.eventbus.api.BusBuilder.builder()' must be InterfaceMethodref constant
Stacktrace:
	at thedarkcolour.kotlinforforge.KotlinModContainer.<init>(KotlinModContainer.kt:40) ~[kotlinforforge-3.6.0-obf.jar%23148!/:3.6.0] {}
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77) ~[?:?] {}
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?] {}
	at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) ~[?:?] {}
	at java.lang.reflect.Constructor.newInstance(Constructor.java:480) ~[?:?] {}
	at thedarkcolour.kotlinforforge.KotlinLanguageProvider$KotlinModTarget.loadMod(KotlinLanguageProvider.kt:46) ~[kotlinforforge-3.6.0-obf.jar%23148!/:3.6.0] {}
	at net.minecraftforge.fml.ModLoader.buildModContainerFromTOML(ModLoader.java:276) ~[fmlcore-1.19.2-43.0.2.jar%23152!/:?] {}
	at net.minecraftforge.fml.ModLoader.lambda$buildMods$29(ModLoader.java:255) ~[fmlcore-1.19.2-43.0.2.jar%23152!/:?] {}
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?] {}
	at java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1850) ~[?:?] {}
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?] {}
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?] {}
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?] {}
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?] {}
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[?:?] {}
	at net.minecraftforge.fml.ModLoader.buildMods(ModLoader.java:257) ~[fmlcore-1.19.2-43.0.2.jar%23152!/:?] {}
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?] {}
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?] {}
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625) ~[?:?] {}
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?] {}
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?] {}
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575) ~[?:?] {}
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260) ~[?:?] {}
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616) ~[?:?] {}
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622) ~[?:?] {}
	at java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627) ~[?:?] {}
	at net.minecraftforge.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:175) ~[fmlcore-1.19.2-43.0.2.jar%23152!/:?] {}
	at net.minecraftforge.client.loading.ClientModLoader.lambda$begin$1(ClientModLoader.java:91) ~[forge-1.19.2-43.0.2-universal.jar%23156!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraftforge.client.loading.ClientModLoader.lambda$createRunnableWithCatch$4(ClientModLoader.java:111) ~[forge-1.19.2-43.0.2-universal.jar%23156!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraftforge.client.loading.ClientModLoader.begin(ClientModLoader.java:91) ~[forge-1.19.2-43.0.2-universal.jar%23156!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.<init>(Minecraft.java:468) ~[client-1.19.2-20220805.130853-srg.jar%23151!/:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:rubidium.mixins.json:core.MixinMinecraftClient,pl:mixin:APP:mixins.ipnext.json:MixinMinecraftClient,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.m_239872_(Main.java:176) ~[client-1.19.2-20220805.130853-srg.jar%23151!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:51) ~[client-1.19.2-20220805.130853-srg.jar%23151!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?] {}
	at net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.lambda$launchService$0(CommonClientLaunchHandler.java:27) ~[fmlloader-1.19.2-43.0.2.jar%2395!/:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) [modlauncher-10.0.8.jar%2382!/:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-10.0.8.jar%2382!/:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-10.0.8.jar%2382!/:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:106) [modlauncher-10.0.8.jar%2382!/:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:77) [modlauncher-10.0.8.jar%2382!/:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-10.0.8.jar%2382!/:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-10.0.8.jar%2382!/:?] {}
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) [bootstraplauncher-1.1.2.jar:?] {}
```

**Comment by Jayryn** [comment]
Could be the InventoryProfilesNext Mod too that's causing the issue but I'm unsure about it.

**EDIT:** Game starts perfectly fine without InventoryProfilesNext -> likely mod related and I already reported it to them.

**Comment by thedarkcolour** [comment]
It is KFF related

**Comment by blackd** [comment]
I've tested with KFF and Inventorio no IPN
Still getting the same problem I think this is the relevant exception 
`Caused by: java.lang.IncompatibleClassChangeError: Method 'net.minecraftforge.eventbus.api.BusBuilder net.minecraftforge.eventbus.api.BusBuilder.builder()' must be InterfaceMethodref constant
`

```
[Render thread/INFO] Backend library: LWJGL version 3.3.1 build 7
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at LAYER PLUGIN/kotlinforforge.obf@3.6.0-obf/thedarkcolour.kotlinforforge.KotlinLanguageProvider$KotlinModTarget.loadMod(KotlinLanguageProvider.kt:46)
	at LAYER PLUGIN/fmlcore@1.19.2-43.0.2/net.minecraftforge.fml.ModLoader.buildModContainerFromTOML(ModLoader.java:276)
	at LAYER PLUGIN/fmlcore@1.19.2-43.0.2/net.minecraftforge.fml.ModLoader.lambda$buildMods$29(ModLoader.java:255)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1850)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at LAYER PLUGIN/fmlcore@1.19.2-43.0.2/net.minecraftforge.fml.ModLoader.buildMods(ModLoader.java:257)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)
	at LAYER PLUGIN/fmlcore@1.19.2-43.0.2/net.minecraftforge.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:175)
	at TRANSFORMER/forge@43.0.2/net.minecraftforge.client.loading.ClientModLoader.lambda$begin$1(ClientModLoader.java:91)
	at TRANSFORMER/forge@43.0.2/net.minecraftforge.client.loading.ClientModLoader.lambda$createRunnableWithCatch$4(ClientModLoader.java:111)
	at TRANSFORMER/forge@43.0.2/net.minecraftforge.client.loading.ClientModLoader.begin(ClientModLoader.java:91)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.client.Minecraft.<init>(Minecraft.java:468)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.client.main.Main.m_239872_(Main.java:176)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.client.main.Main.main(Main.java:51)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at MC-BOOTSTRAP/fmlloader@1.19.2-43.0.2/net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.lambda$launchService$0(CommonClientLaunchHandler.java:27)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.Launcher.run(Launcher.java:106)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.Launcher.main(Launcher.java:77)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23)
	at cpw.mods.bootstraplauncher@1.1.2/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141)
Caused by: java.lang.IncompatibleClassChangeError: Method 'net.minecraftforge.eventbus.api.BusBuilder net.minecraftforge.eventbus.api.BusBuilder.builder()' must be InterfaceMethodref constant
	at LAYER PLUGIN/kotlinforforge.obf@3.6.0-obf/thedarkcolour.kotlinforforge.KotlinModContainer.<init>(KotlinModContainer.kt:40)
	... 46 more
```

**Comment by thedarkcolour** [comment]
Just checked Forge changelog. Should work with 43.0.0, but nothing past that version

**Comment by Jayryn** [comment]
> Just checked Forge changelog. Should work with 43.0.0, but nothing past that version

I'll try that later, thanks for the reminder.
May 43.0.2 also be an option to add later on to support?

**Comment by thedarkcolour** [comment]
Fixed in Kotlin for Forge 3.7.1

**Comment by Jayryn** [comment]
> Fixed in Kotlin for Forge 3.7.1

Thanks
